### PR TITLE
Merge Acquire and Late Prefetch

### PR DESCRIPTION
**Pull request by cailuoshan**
merge Acquire -> Late Prefetch in MSHR;
remove w_grantack and use inflightgrant idx as sourceD_sink

### CODE REVIEW

File: src/main/scala/coupledL2/GrantBuffer.scala [hunk 1]
```diff
@@ -31,14 +31,19 @@ import coupledL2.utils.{XSPerfAccumulate, XSPerfHistogram, XSPerfMax}
 class InflightGrantEntry(implicit p: Parameters) extends L2Bundle {
   val set   = UInt(setBits.W)
   val tag   = UInt(tagBits.W)
-  val sink  = UInt(mshrBits.W)
 }
 
 class TaskWithData(implicit p: Parameters) extends L2Bundle {
   val task = new TaskBundle()
   val data = new DSBlock()
```
Comment (Ivyfeather) [inline src/main/scala/coupledL2/GrantBuffer.scala:38]: remove TaskWithData

File: src/main/scala/coupledL2/GrantBuffer.scala [hunk 2]
```diff
@@ -104,12 +106,50 @@ class GrantBuffer(implicit p: Parameters) extends L2Module {
 //    (next_beat, next_beatsOH)
 //  }
 
+  val grantQueue = Module(new Queue(new GrantQueueTask(), entries = mshrsAll))
+  val inflight_grant = RegInit(VecInit(Seq.fill(grantBufInflightSize){
```
Comment (Ivyfeather) [inline src/main/scala/coupledL2/GrantBuffer.scala:110]: rename as inflightGrant
use this commit to change to Camel-Case

File: src/main/scala/coupledL2/GrantBuffer.scala [hunk 3]
```diff
@@ -175,17 +218,18 @@ class GrantBuffer(implicit p: Parameters) extends L2Module {
...
     entry.valid := true.B
     entry.bits.set    := io.d_task.bits.task.set
     entry.bits.tag    := io.d_task.bits.task.tag
-    entry.bits.sink   := io.d_task.bits.task.mshrId
+  } .elsewhen (io.d_task.fire && io.d_task.bits.task.mergeA) {
+    // if mergeA, alloc an entry for merge grant 
+    val entry = inflight_grant(inflight_insertIdx)
+    entry.valid := true.B
+    entry.bits.set := io.d_task.bits.task.set
+    entry.bits.tag := io.d_task.bits.task.tag
```
Comment (Ivyfeather) [inline src/main/scala/coupledL2/GrantBuffer.scala:232]: is the logic is `when` and `elsewhen` **the same**?

**Review by Ivyfeather — APPROVED** [review]

File: src/main/scala/coupledL2/GrantBuffer.scala (hunk 1 above)
Comment (cailuoshan) [reply src/main/scala/coupledL2/GrantBuffer.scala:38]: TaskWithData should be used in io_d_task

File: src/main/scala/coupledL2/GrantBuffer.scala (hunk 3 above)
Comment (cailuoshan) [reply src/main/scala/coupledL2/GrantBuffer.scala]: yes, I have merged them.

**Pull request merged**